test/e2e_node/dra_test.go

```diff
@@ -111,51 +113,352 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
 		})
 
 		ginkgo.It("must keep pod in pending state if NodePrepareResources times out", func(ctx context.Context) {
-			ginkgo.By("set delay for the NodePrepareResources call")
-			kubeletPlugin.Block()
-			pod := createTestObjects(ctx, f.ClientSet, getNodeName(ctx, f), f.Namespace.Name, "draclass", "external-claim", "drapod")
+			kubeletPlugin.WithBlockedNodePrepareResources(func() (*v1.Pod, func()) {
```

Comment:
I find it hard to follow the flow of the test because of this callback approach. Can we do it without those `With*` calls?

This could read like this instead:
```
unblock := kubeletPlugin.BlockNodePrepareResources()
defer unblock()

// code which runs while blocked
```

I'm aware that this puts a little more responsibility on the caller. But I think the additional clarity is worth it.

Reply:
With the current approach it's more readable as it's easier to see the code that runs when `[Un]PrepareResources` is blocked. It adds a nice structure to the code from my point of view. Not sure I want to go back to the plain less unstructured code. Will do only if you insist. Please, confirm.

Reply:
Perhaps you as the author understood this. I as reviewer didn't not find it readable.

Please change it.

Reply:
changed.